src/actor/npc/ac_npc_shop_master.c

```diff
@@ -0,0 +1,2677 @@
+    stack_1.z = ((f32)((int)player_actor->forward_ut_pos.z / mFI_UT_WORLDSIZE_Z) + 0.5f) * mFI_UT_WORLDSIZE_Z_F;
+
+    pos_8.y = atans_table(stack_1.z - player_actor->actor_class.world.position.z, stack_1.x - player_actor->actor_class.world.position.x);
+
+    GET_PLAYER_ACTOR_NOW()->Set_force_position_angle_proc(gamePT, &player_actor->actor_class.world.position, &pos_8, mPlayer_FORCE_POSITION_ANGLE_ROTY);
+}
+
+static void aNSC_set_talk_info_show_item(NPC_SHOP_MASTER_ACTOR *actor, mActor_name_t item) {
+    aNSC_set_item_str(actor, item);
+    aNSC_set_player_angle();
+    mDemo_Set_camera(1);
+    mDemo_Set_msg_num(aNSC_get_msg_no(aNSC_MSG_SHOW_ITEM));
+}
+
+static void aNSC_set_talk_info_show_cloth(NPC_SHOP_MASTER_ACTOR *actor, mActor_name_t cloth) {
+    PositionAngle pa;
+    aNSC_set_item_str(actor, cloth);
+    aNSC_set_player_angle();
+    mDemo_Set_camera(0xd);
+    mDemo_Set_msg_num(aNSC_get_msg_no(aNSC_MSG_SHOW_CLOTH));
+}
+
+static void aNSC_set_talk_info_sell_item(NPC_SHOP_MASTER_ACTOR *actor, mActor_name_t p2) {
+    mActor_name_t item = actor->sell_item;
+    u8 camera_type;
+    int msg_no = aNSC_MSG_BUY_CANCEL;
+
+    aNSC_set_item_str(actor, p2);
+    aNSC_set_player_angle();
+
+    if (ITEM_NAME_GET_TYPE(item) == NAME_TYPE_FTR0 || ITEM_NAME_GET_TYPE(item) == NAME_TYPE_FTR1 || (ITM_UMBRELLA_START <= item && item <= ITM_UMBRELLA_END - 1)) {
+        camera_type = mDemo_TYPE_EVENTMSG;
+    } else {
+        camera_type = mDemo_TYPE_SCROLL;
+    }
+
+    if (item >= ITM_RED_PAINT && item <= ITM_BROWN_PAINT) {
+        if (mLd_PlayerManKindCheck() == FALSE) {
+            msg_no = aNSC_MSG_SELL_PAINT;
+        } else {
+            msg_no = aNSC_MSG_SELL_PAINT_FOREIGN;
+        }
+    } else {
+        if (item >= ITM_PAPER_START && item <= ITM_PAPER_END - 1) {
+            msg_no = aNSC_MSG_SELL_PAPER;
+        }
+    }
+
+
+    mDemo_Set_camera(camera_type);
+    mDemo_Set_msg_num(aNSC_get_msg_no(msg_no));
+}
+
+static void aNSC_set_talk_info_message_ctrl() {
+    mDemo_Set_msg_num(aNSC_get_msg_no(aNSC_MSG_INTERACT_START));
+}
+
+static void aNSC_set_talk_info_message_ctrl_aprilfool(ACTOR *actorx, GAME *game) {
+    int msg_num = Common_Get(clip).aprilfool_control_clip->get_msg_num_proc(SP_NPC_SHOP_MASTER, 0);
+    mDemo_Set_msg_num(msg_num);
+}
+
+static void aNSC_set_talk_info_message_ctrl_tokubai() {
+    int msg_num = aNSC_get_msg_no(aNSC_MSG_GREET_DEPART);
+    mDemo_Set_msg_num(msg_num);
+}
+
+static int aNSC_message_ctrl(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *game) {
+    PLAYER_ACTOR *player = GET_PLAYER_ACTOR_GAME(game);
+    int res = 0;
+
+    if (player != NULL && CLIP(shop_design_clip) != NULL) {
+        if (player->item_in_front == EXIT_DOOR1) {
+            aNSC_setupAction(shop_master, game, aNSC_ACTION_EXIT);
+            res = TRUE;
+        } else {
+            ACTOR * actorx = (ACTOR *)shop_master;
+            int action = 0;
```

Comment:
Another case where this needs to be converted to the aforementioned action enum. I'll stop commenting on these after this comment, but they all need to be changed.